pandas/tests/series/methods/test_astype.py

```diff
@@ -97,6 +97,20 @@ def test_astype_dict_like(self, dtype_class):
 
 
 class TestAstype:
+    def test_astype_mixed_object_to_dt64tz(self):
+        # pre-2.0 this raised ValueError bc of tz mismatch
+        # xref GH#32581
+        ts = Timestamp("2016-01-04 05:06:07", tz="US/Pacific")
+        ts2 = ts.tz_convert("Asia/Tokyo")
+
+        ser = Series([ts, ts2], dtype=object)
+        res = ser.astype("datetime64[ns, Europe/Brussels]")
```

Comment:
Whats the error message if we try this when an element is not convertable e.g. `ser = Series([ts, object], dtype=object)`

Reply:
`ValueError: could not convert string to Timestamp`